velox/common/caching/AsyncDataCache.cpp

```diff
@@ -676,6 +691,43 @@ bool AsyncDataCache::makeSpace(
   return false;
 }
 
+uint64_t AsyncDataCache::shrink(uint64_t targetBytes) {
+  VELOX_CHECK_GT(targetBytes, 0);
+
+  const uint64_t minBytesToEvict = 8UL << 20;
+  uint64_t evictedBytes{0};
+  for (int attempt = 0; attempt < 1; ++attempt) {
+    for (int shard = 0; shard < shards_.size(); ++shard) {
+      if (attempt > 0 && (ssdCache_ != nullptr) &&
+          ssdCache_->writeInProgress()) {
```

Comment:
Curious, how do we test SSD-specific logic?

Reply:
Shrink doesn't change the ssd cache behavior. It mainly adds a sleep if there is on-going ssd write. Shrink doesn't trigger ssd write itself. We could have a test to trigger ssd writes to race with cache shrink. Also we could expose ssd write triggering metrics to verify if ssd has been triggered.